### DISCUSSION TITLE
m2e is not copying resources anymore

### DISCUSSION THREAD

**Discussion by ahoehma** [eclipse-m2e/m2e-core#1142] [General]
Not sure but I noticed since a while that in my eclipse the resources/test-resources are not copied/filtered etc. anymore .
When I run cmdline maven everything is fine.

![image](https://user-images.githubusercontent.com/548214/208253116-c7c9ddd0-5e0e-48b0-b9c4-348b16a2da17.png)

**Comment by HannesWell** [comment]
I noticed that recently as well that it did not Work recently.
Debugging revealed that, at least in my case, the reason was that M2E failed to load the Mojo or realm with an Exception.
Since it only happened ocasionally I assumed that it was only a local short-term (Network?) problem.
But maybe that was a false assumption.

Do you see any related entries in the Error-Log View?

**Reply by ahoehma** [reply]
I find nothing in the error-log :(

**Reply by laeubi** [reply]
Do you have a project to reproduce the issue?

**Reply by ahoehma** [reply]
Not really. Its a big reactor and under closed source. May you can tell me if you have some specific maven-plugins in mind which could be maybe wrong configured, like the resource-plugin etc. But I can swear it was working :)

**Reply by ahoehma** [reply]
I switched in the pom editor to "effective pom" and grabbed these "resources" stuff out of it:

```xml
<resources>
      <resource>
        <directory>D:\Dev\git\coc\spice-configuration-cluster\datamodel-adapter\src\main\resources</directory>
      </resource>
    </resources>
    <testResources>
      <testResource>
        <directory>D:\Dev\git\coc\spice-configuration-cluster\datamodel-adapter\src\test\resources</directory>
      </testResource>
    </testResources>
```
```xml
<plugin>
        <artifactId>maven-resources-plugin</artifactId>
        <version>3.3.0</version>
        <executions>
          <execution>
            <id>default-testResources</id>
            <phase>process-test-resources</phase>
            <goals>
              <goal>testResources</goal>
            </goals>
          </execution>
          <execution>
            <id>default-resources</id>
            <phase>process-resources</phase>
            <goals>
              <goal>resources</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
```

looks pretty normal to me

**Reply by ahoehma** [reply]
I have also a lifecycle mapping in the module 

```xml
<plugin>
          <groupId>org.eclipse.m2e</groupId>
          <artifactId>lifecycle-mapping</artifactId>
          <version>1.0.0</version>
          <configuration>
            <lifecycleMappingMetadata>
              <pluginExecutions>
                <pluginExecution>
                  <pluginExecutionFilter>
                    <groupId>com.github.victools</groupId>
                    <artifactId>jsonschema-maven-plugin</artifactId>
                    <versionRange>[4.18.0,)</versionRange>
                    <goals>
                      <goal>generate</goal>
                    </goals>
                  </pluginExecutionFilter>
                  <action>
                    <execute>
                      <runOnConfiguration>true</runOnConfiguration>
                      <runOnIncremental>false</runOnIncremental>
                    </execute>
                  </action>
                </pluginExecution>
              </pluginExecutions>
            </lifecycleMappingMetadata>
          </configuration>
        </plugin>
```

**Comment by tehrnhoefer** [comment]
FWIW, I am running into the same problem (took me quite a while to realize what's going on).
Running maven on the command line successfully copies resources, not so much in Eclipse.

  M2E - Maven Integration for Eclipse	2.1.2.20221130-2239	org.eclipse.m2e.feature.feature.group	Eclipse.org - m2e
  Eclipse IDE for Java Developers	4.26.0.20221201-1200	epp.package.java	Eclipse Packaging Project

**Reply by laeubi** [reply]
> I have also a lifecycle mapping in the module

You disable the goal with this configuration on code changes, so it only runs if you update the project.

**Reply by ahoehma** [reply]
I know ... that's here what I want. Alt+F5 is enough for the jsonschema plugin.
But this is not the main topic in the issue here.
I see for **all** resources (src/main/resources, src/test/resources) that the are not copied/filtered etc. It seems the whole resource handling is not working in m2e

**Reply by laeubi** [reply]
Then maybe a minimal project with expected outcome would help here? Its a bit confusing to mention  lifecycle mapping if they are not part of the problem. 
You still should be able to open the life-cycle-mapping view to see if the resources-plugin has appropriate mappings defined.

**Reply by ahoehma** [reply]
I was just sharing the particular "effective pom" of the project where I really see the missing resources after "Alt-F5". Just as let you know all the "special" things from my point of view. I will try to find a more minimal project to see the effect. Theoretically: the m2e tests should show such "essential" fails. If not then it must be something only in my workspace. Hard challenge :)

**Reply by tehrnhoefer** [reply]
I reverted back to   M2E - Maven Integration for Eclipse	2.0.5.20220912-1211
...and the problem disappears
updating to     M2E - Maven Integration for Eclipse	2.0.6.20221122-2114 
and the problem is back

**Reply by HannesWell** [reply]
> I reverted back to M2E - Maven Integration for Eclipse 2.0.5.20220912-1211 ...and the problem disappears updating to M2E - Maven Integration for Eclipse 2.0.6.20221122-2114 and the problem is back

These version refer to the 2.0.5 and 2.1.0 release of M2E.
So it is probably something in between (or just the installation process triggers some 'deeper' re-fresh or alike).

In my use cases the resources-plugin actually works fine.
Can you try the latest snapshot version? You can find it here: https://download.eclipse.org/technology/m2e/snapshots/2.2.0/

**Reply by tehrnhoefer** [reply]
I just tried the snapshot, unsurprisingly (as with any version since 2.1.0) it still doesn't copy the resources :(

**Reply by KaRkY** [reply]
I had the same problem and found out that if you have ".mvn" folder at the root of the project it fails to copy resources.

**Reply by ahoehma** [reply]
@KaRkY aha ... I have .mvn in my root project folder :)

**Reply by egeorges92** [reply]
> I had the same problem and found out that if you have ".mvn" folder at the root of the project it fails to copy resources.

Thank you for the tips. It's solved my issue. It stuck me for a while :-(

**Reply by ahoehma** [reply]
@egeorges92 do you removed the .mvn? Just asking. Because this is not an option for me I need .mvn for extensions.xml.

**Reply by egeorges92** [reply]
Hello @ahoehma , yes, I remove all of these {.mvn, mvnw, mvnw.cmd).
I don’t use enforcer plugin, it’s not an issue for me.
The only solution is to use the old way, have a look at Maven documentation : https://maven.apache.org/configure.html

**Reply by ahoehma** [reply]
@egeorges92 I'm using .mvn/extensions.xml to work with "maven-git-versioning-extension".
